src/Util.js

```diff
@@ -37,40 +37,65 @@ function encodeString(s, d) {
   return data;
 }
 
-function decodePacket(data) {
-  if(!data.length) return [];
+function decodePackets(data) {
+  if (!data.length) return [];
   const buf = [];
+  let leftover;
   let idx = 0;
-  while(idx < data.length) {
-    let len;
+  let bbuf = [];
+loop:
+  while (idx < data.length) {
     let b = data[idx++];
-    switch(true) {
-      case (b & 192) === 128:
-        len = ((b & 63) << 8) + data[idx++];
+    let len = b;
+    switch (true) {
+      case b === 0x00:
+        buf.push(bbuf);
+        bbuf = [];
+        continue loop;
+      case (b & 0x80) === 0x00:
         break;
-      case (b & 224) === 192:
-        len = ((b & 31) << 8) + data[idx++];
-        len = (len << 8) + data[idx++];
+      case (b & 0xC0) === 0x80:
+        len &= ~0xC0;
+        len <<= 8;
+        len += data[idx++];
         break;
-      case (b & 240) === 224:
-        len = ((b & 15) << 8) + data[idx++];
-        len = (len << 8) + data[idx++];
-        len = (len << 8) + data[idx++];
+      case (b & 0xE0) === 0xC0:
+        len &= (~0xE0);
+        len <<= 8;
+        len += data[idx++];
```

Comment:
Can we keep these operations as they were, it makes the code much more compact, and I find it more understandable. One suggestion I would have is we keep it bitwise operations only, no addition, again just for clarity.
` len = (len << 8) | data[idx++];`

Reply:
Yes, I definitely agree. I pulled it all apart so that it matches the python example implementation almost completely (from `readLen` here: https://wiki.mikrotik.com/wiki/Manual:API#Example_client)